Clippy throws `needless_lifetimes` on components with lifetime parameter

**Describe the bug**
In leptos 0.4, I write a component taking a reference like this:

```rust
#[component]
fn FlakeView<'a>(flake: &'a Flake) -> impl IntoView {
```

The same is true for leptos 0.5 ― without explicit `'a`, the compiler will complain `error[E0106]: missing lifetime specifier`.

However in 0.5 (but not in 0.4) clippy started complaining about this lifetime parameter being "needless" when it is in fact needed. 

```
warning: the following explicit lifetimes could be elided: 'a
  --> src/app/flake.rs:87:14
   |
87 | fn FlakeView<'a>(flake: &'a Flake) -> impl IntoView {
   |              ^^          ^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
   = note: `#[warn(clippy::needless_lifetimes)]` on by default
help: elide the lifetimes
   |
87 - fn FlakeView<'a>(flake: &'a Flake) -> impl IntoView {
87 + fn FlakeView(flake: &Flake) -> impl IntoView {
```

Did something change in the component macro?


**Leptos Dependencies**

```toml
leptos = { version = "0.5", features = ["serde", "nightly"] }
leptos_meta = { version = "0.5", features = ["nightly"] }
leptos_router = { version = "0.5", features = ["nightly"] }
leptos_query = "0.3"
```


**To Reproduce**

Run `nix develop -c cargo clippy` in https://github.com/juspay/nix-browser/pull/78 (commit [bf90433](https://github.com/juspay/nix-browser/pull/78/commits/bf904333f6ea962ed30ccc184be1dfb265160045))

**Expected behavior**

`cargo clippy` should succeed without warnings.

**Screenshots**

![image](https://github.com/leptos-rs/leptos/assets/3998/144a6dd8-d2d1-4e38-abee-6f52db7f728c)

--- a/leptos_macro/src/component.rs
+++ b/leptos_macro/src/component.rs
@@ -511,6 +511,7 @@ impl ToTokens for Model {
             #[doc = ""]
             #component_fn_prop_docs
             #[allow(non_snake_case, clippy::too_many_arguments)]
+            #[allow(clippy::needless_lifetimes)]
             #tracing_instrument_attr
             #vis fn #name #impl_generics (
                 #props_arg

